plugins/imgur_a.js

```diff
@@ -42,32 +46,41 @@ hoverZoomPlugins.push({
                     
                     switch (view) {
                         case 'signin':
-                            return;
+                        return;
                         case 'a': // album view:
                         case 'gallery':
-                            var anchor = matches[3];
-                            data.hoverZoomGallerySrc = [];
-                            data.hoverZoomGalleryCaption = [];
 
                             // Future alternative: https://imgur.com/ajaxalbums/getimages/{hash}/hit.json?all=true
                             var albumUrl = 'https://api.imgur.com/3/album/' + hash + '.json';
                             $.ajax(albumUrl, {headers: {"Authorization": "Client-ID 1d8d9b36339e0e2"}}).done(function (imgur) {
                                 if (imgur.error) {
                                     data.hoverZoomSrc = createUrls(hash);
                                     res.push(link);
-                                } else {
+                                } 
+                                //todo: Possible faster implementation later on for single images - leave commented for now
+                                /*
+                                else if(imgur.data.images_count === 1){
+                                    data.hoverZoomSrc = createUrls(imgur.data.images[0].id);
+                                    link.addClass('hoverZoomLink');
+                                    res.push(link);
+                                }
+                                */
+                                else {
+                                    var anchor = matches[3];
+                                    data.hoverZoomGallerySrc = [];
+                                    data.hoverZoomGalleryCaption = [];
                                     imgur.data.images.forEach(function (img, index) {
                                         var urls = [img.link],
-                                            caption = img.title,
-                                            alreadyAdded = false;
+                                        caption = img.title,
+                                        alreadyAdded = false;
                                         for (var i=0, l=data.hoverZoomGallerySrc.length; i<l; i++) {
                                             if (data.hoverZoomGallerySrc[i].indexOf(urls[0]) != -1) {
                                                 alreadyAdded = true;
                                                 break;
                                             }
                                         }
                                         if (!alreadyAdded) {
-                                            if (caption != '' && img.description != '') {
+                                            if (caption != '' && caption != 'null' && img.description != '') {
```

Comment:
if caption is string "null", shouldn't you clear it first before appending description? otherwise you'll get text "nullDescription"